packages/neon-core/src/sc/ContractParam.ts

```diff
@@ -64,15 +67,23 @@ export class ContractParam implements NeonObject<ContractParamLike> {
     });
   }
 
-  public static publicKey(value: string): ContractParam {
-    return new ContractParam({ type: ContractParamType.PublicKey, value });
+  public static publicKey(value: string | HexString): ContractParam {
+    return new ContractParam({
+      type: ContractParamType.PublicKey,
+      value: value instanceof HexString ? value : HexString.fromHex(value),
+    });
   }
 
   /**
-   * Creates a Hash160 ContractParam. This is used for containing a scriptHash. Do not reverse the input if using this format.
+   * Creates a Hash160 ContractParam. This is used for containing a scriptHash. Value field will be a HexString.
+   * Do not reverse the input if using this format.

```

Comment:
I find this a confusing remark, but that probably has to do with `neon-js` reversing byte arrays in many places I don't expect it to. For clarification purposes only; can you give the rules when `neon-js` does and does not expect a reversed hex string?

Reply:
Usually it is big endian. In this case, a Hash160's value is parsed as bigendian on C# side while a ByteArray is not. But the rule of thumb in neon-js has always been to keep moving big endian around.

Reply:
Sorry it is still ambiguous for me. What do you call big endian in this context? I ask this because the NEO C# team has this crazy idea that adding a hexadecimal prefix (`0x`) suddenly changes the endianness of the data. I couldn't blame you if you follow their convention, but because it deviates from the standard I'm still confused what it means

> But the rule of thumb in neon-js has always been to keep moving big endian around.

Does this mean that you do not reverse Hash160/Hash256 types (`UInt160/UInt256` in C#) but reverse everything else?

Reply:
The reference case is usually the ByteArray used in C#. This is little-endian. So we extrapolate from this case to all other instances. Im mentioning the reversing because in NEO2, the convention was `ContractParam.byteArray(reverseHex(getScriptHashFromAddress(address)))`. 

In `neon-js` the `0x` prefix is usually discarded. Im aware of the convention but currently not adopting it.

What is your definition of the standard?

Reply:
> What is your definition of the standard?

In general my definition is that the only time I care about endianness is when reading primitive types from a byte array where the primitive type is 2 or more bytes in size (e.g. short, int, long etc). Putting this in the NEO context and speaking from my Python project (which is a full port from C#), there are only 2 places in the whole code base where I reverse the data and they only happen for the `UInt160` and `UInt256` types.
* from string - taking a UInt160/UInt256 string representation produced by the C# code
* to string - making sure it produces the same string representation **as** the C# code (because many other tools expect this format)

Everything else internally can be said to be little endian. So my binary reader/writer classes read all data by default in little endian format. Which is why I previously mentioned that everything in `neon-js` seems to be in reversed format for me, because your representation seems to be big endian by default. :)